### PR TITLE
test: inital commit

### PR DESCRIPTION
**Pull request by nalmabrcom**
## Cypress Test Library for Theia
Cypress comes with its own API for creating custom commands and overwriting existing commands. The built-in Cypress commands use the very same API that’s defined below.
A great place to define commands (or functions) is in the `cypress/support/commands.js` file since it is loaded before any test files are evaluated via an import statement in supportFile (`cypress/support/index.js` by default).
But we want not just to have those commands in one place (i.e. project with test suites), but import them as a package to any project and use custom commands. 

Also, I checked every dependency in the project with [ClearlyDefined](https://bit.ly/39VGi2g)

Signed-off-by: Maryna Nalbandian <maryna.nalbandian@broadcom.com>

### CODE REVIEW

**Review by filipkroupa — COMMENTED** [review]
lgtm

File: tests/src/debug.ts [hunk 1]
```diff
@@ -0,0 +1,693 @@
...
+     * Read and write JSON file
+     * Copy from test_files/project/dap/ specific launch json file
+     * and paste to .theia/launch.json
+     * @example cy.readAndWriteJsonFile('TEST')
+     */
+    readAndWriteJsonFile(expression: string): Chainable<any>;
+  }
+}
+Cypress.Commands.add('readAndWriteJsonFile', (expression: string) => {
+  cy.readFile(`test_files/project/dap/${expression}.launch.json`).then((content) => {
```
Comment (grianbrcom) [inline tests/src/debug.ts:37]: You have hardcoded `test_files` folder here, but it not appears in the PR. 
Will it be a separate change?

File: tests/src/debug.ts [hunk 2]
```diff
@@ -0,0 +1,693 @@
...
+     */
+    clickOnContinue(): Chainable<any>;
+  }
+}
+Cypress.Commands.add('clickOnContinue', () => {
+  cy.get(`[id^="debug:toolbar"] .debug-action.theia-debug-continue`)
+    .should('not.have.class', 'theia-mod-disabled')
+    .click()
+    .click()
+    .wait(2000);
```
Comment (grianbrcom) [inline tests/src/debug.ts:458]: You can extract it into a separate method named `theiaClick`.

**Review by grianbrcom — APPROVED** [review]
Looks good.
I'm not sure about the `LICENSE` file in the tests folder. Maybe it will be better to put it into the repo root. Otherwise, it's not clear what the license has the whole repo?

File: tests/src/debug.ts (hunk 1 above)
Comment (nalmabrcom) [reply tests/src/debug.ts:37]: `test_files` folder will appear in the test suite project (i.e. in COBOL LS Functional Test Suite), but it's in dev stage at the moment.

File: tests/src/debug.ts (hunk 2 above)
Comment (nalmabrcom) [reply tests/src/debug.ts:458]: Nice idea 👍

**Pull request merged**